ppmclibs/tinycv_impl.cc

```diff
@@ -197,11 +181,15 @@ std::vector<int> search_TEMPLATE(const Image* scene, const Image* object,
     int scene_width = scene_bottom_x - scene_x;
     int scene_height = scene_bottom_y - scene_y;
 
-    Mat scene_copy = scene->prep();
-    Mat object_copy = object->prep();
+    Mat scene_roi = Mat(scene->img, Rect(scene_x, scene_y, scene_width, scene_height));
+    Mat object_roi = Mat(object->img, Rect(x, y, width, height));
 
-    Mat scene_roi(scene_copy, Rect(scene_x, scene_y, scene_width, scene_height));
-    Mat object_roi(object_copy, Rect(x, y, width, height));
+    for (auto t : { &scene_roi, &object_roi }) {
+        Mat n;
+        GaussianBlur(*t, n, Size(3, 3), 0, 0);
```

Comment:
make that 'the ROI only'. It's especially tricky if the ROI is only some pixels large.